### PR TITLE
Update release process to use templating

### PR DESCRIPTION
**Pull request by sjanota**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: guidelines/releases-guidelines/02-kyma-release-process.md [hunk 1]
```diff
@@ -33,77 +33,15 @@ To prepare a release:
...
-         - pre-rel14-kyma-integration
-         - pre-rel14-kyma-gke-integration
-         - pre-rel14-kyma-gke-upgrade
-         - pre-rel14-kyma-gke-central-connector
-         - pre-rel14-kyma-artifacts
-         - pre-rel14-kyma-installer
-         - pre-rel14-kyma-gke-minio-gateway
-   ```
-
-### Remove old release jobs
```
Comment (suleymanakbas91) [inline guidelines/releases-guidelines/02-kyma-release-process.md]: You can remove this step from the preparation list above.

Comment (sjanota) [reply guidelines/releases-guidelines/02-kyma-release-process.md]: Done

File: guidelines/releases-guidelines/02-kyma-release-process.md [hunk 2]
```diff
@@ -29,81 +29,19 @@ To prepare a release:
...
-
-1. After adding new release jobs, remove the old ones. Remember to leave jobs for three latest releases. For example, during the preparation for the 1.4 release, add `pre-rel14` jobs and remove all `pre-rel11` jobs. Make sure that the only defined jobs are those with `pre-rel12`, `pre-rel13`, and `pre-rel14` prefixes.
-
-2. Additionally, remove the unsupported `tester.go` file and all references to it. For example, when releasing 1.4, `Release11 SupportedRelease = "release-1.1"` should be removed.
-
-3. Merge the PR to the current master branch.
+1. Navigate to the `test-infra` repository.
+2. Open `templates/config.yaml`
+3. Change `global.releases` to contain new release. Also remove oldest release on the list.
```
Comment (bszwarc) [inline guidelines/releases-guidelines/02-kyma-release-process.md]:
```suggestion
3. Add the new release to `global.releases`. Remove the oldest release on the list.
```

File: guidelines/releases-guidelines/02-kyma-release-process.md [hunk 3]
```diff
@@ -29,81 +29,19 @@ To prepare a release:
...
+4. Change `global.nextRelease` to contain version for future release.
```
Comment (bszwarc) [inline guidelines/releases-guidelines/02-kyma-release-process.md]:
```suggestion
4. Set `global.nextRelease` to the future release version. 
```

File: guidelines/releases-guidelines/02-kyma-release-process.md [hunk 4]
```diff
@@ -29,81 +29,19 @@ To prepare a release:
...
-2. Additionally, remove the unsupported `tester.go` file and all references to it. For example, when releasing 1.4, `Release11 SupportedRelease = "release-1.1"` should be removed.
-
-3. Merge the PR to the current master branch.
+1. Navigate to the `test-infra` repository.
+2. Open `templates/config.yaml`
+3. Change `global.releases` to contain new release. Also remove oldest release on the list.
+4. Change `global.nextRelease` to contain version for future release. 
+5. Run `dep ensure -v --vendor-only` in the `development/tools` directory to install dependencies.
+5. Run `go run development/tools/cmd/rendertemplates/main.go --config templates/config.yaml` in the root of the repository to generate jobs.
+6. Run `go test ./development/tools/jobs/...` in the root of the repository. If anything is red fix it using these guidelines:
```
Comment (bszwarc) [inline guidelines/releases-guidelines/02-kyma-release-process.md]:
```suggestion
6. Run `go test ./development/tools/jobs/...` in the root of the repository. If any of the tests is marked red, fix it using these guidelines: 
```

File: guidelines/releases-guidelines/02-kyma-release-process.md [hunk 5]
```diff
@@ -29,81 +29,19 @@ To prepare a release:
...
+1. Navigate to the `test-infra` repository.
+2. Open `templates/config.yaml`
+3. Change `global.releases` to contain new release. Also remove oldest release on the list.
+4. Change `global.nextRelease` to contain version for future release. 
+5. Run `dep ensure -v --vendor-only` in the `development/tools` directory to install dependencies.
+5. Run `go run development/tools/cmd/rendertemplates/main.go --config templates/config.yaml` in the root of the repository to generate jobs.
+6. Run `go test ./development/tools/jobs/...` in the root of the repository. If anything is red fix it using these guidelines: 
+  * For release tests using `GetKymaReleasesSince` with a release that is no longer supported change it to use `GetAllKymaReleases`.
+  * For release tests using `GetKymaReleasesUntil` with a release that is no longer supported simply remove that part of the test.
+7. When tests are green commit all jobs and you just got yourself new release jobs.
```
Comment (bszwarc) [inline guidelines/releases-guidelines/02-kyma-release-process.md]:
```suggestion
7. If tests are green, commit all jobs. The new release jobs are ready.
```

File: guidelines/releases-guidelines/02-kyma-release-process.md [hunk 6]
```diff
@@ -29,81 +29,19 @@ To prepare a release:
...
-
-3. Merge the PR to the current master branch.
+1. Navigate to the `test-infra` repository.
+2. Open `templates/config.yaml`
+3. Change `global.releases` to contain new release. Also remove oldest release on the list.
+4. Change `global.nextRelease` to contain version for future release. 
+5. Run `dep ensure -v --vendor-only` in the `development/tools` directory to install dependencies.
+5. Run `go run development/tools/cmd/rendertemplates/main.go --config templates/config.yaml` in the root of the repository to generate jobs.
+6. Run `go test ./development/tools/jobs/...` in the root of the repository. If anything is red fix it using these guidelines: 
+  * For release tests using `GetKymaReleasesSince` with a release that is no longer supported change it to use `GetAllKymaReleases`.
```
Comment (bszwarc) [inline guidelines/releases-guidelines/02-kyma-release-process.md]:
```suggestion
  * For release tests using `GetKymaReleasesSince` with a release that is no longer supported, change the method to `GetAllKymaReleases`.
```

File: guidelines/releases-guidelines/02-kyma-release-process.md [hunk 7]
```diff
@@ -29,81 +29,19 @@ To prepare a release:
...
+  * For release tests using `GetKymaReleasesUntil` with a release that is no longer supported simply remove that part of the test.
```
Comment (bszwarc) [inline guidelines/releases-guidelines/02-kyma-release-process.md]:
```suggestion
  * For release tests using `GetKymaReleasesUntil` with a release that is no longer supported, remove the part of the test which includes the method.
```

File: guidelines/releases-guidelines/02-kyma-release-process.md [hunk 8]
```diff
@@ -29,81 +29,19 @@ To prepare a release:
...
-   ```
-
-### Remove old release jobs
-
-1. After adding new release jobs, remove the old ones. Remember to leave jobs for three latest releases. For example, during the preparation for the 1.4 release, add `pre-rel14` jobs and remove all `pre-rel11` jobs. Make sure that the only defined jobs are those with `pre-rel12`, `pre-rel13`, and `pre-rel14` prefixes.
-
-2. Additionally, remove the unsupported `tester.go` file and all references to it. For example, when releasing 1.4, `Release11 SupportedRelease = "release-1.1"` should be removed.
-
-3. Merge the PR to the current master branch.
+1. Navigate to the `test-infra` repository.
```
Comment (bszwarc) [inline guidelines/releases-guidelines/02-kyma-release-process.md]:
```suggestion
1. Go to the `test-infra` repository.
```

**Pull request merged**